### PR TITLE
Add Config.read_config/1

### PR DESCRIPTION
**Pull request by josevalim**

### CODE REVIEW

File: lib/elixir/lib/config.ex [hunk 1]
```diff
@@ -194,6 +194,31 @@ defmodule Config do
     |> put_config()
   end
 
+  @doc """
+  Reads the configuration for the given root key.
+
+  This function only reads the configuration from a previous
+  `config/2` pr `config/3` call. If `root_key` points to an
```
Comment (voughtdq) [inline lib/elixir/lib/config.ex:201]: minor typo here "pr" -> "or"

File: lib/logger/lib/logger.ex [hunk 2]
```diff
@@ -226,13 +226,30 @@ defmodule Logger do
...
+        filter_default: :log,
+        formatter: &Logger.default_formatter/0,
+        level: :all,
+        module: :logger_std_h
+      ]
+
+  The `:config` customizes a specific handler module. The default handler
+  is [`:logger_std_h`](`:logger_std_h`), which logs to standard IO, and you
+  call find all relevant configuration in its modulee documentation, including
+  information overload protection.
```
Comment (voughtdq) [inline lib/logger/lib/logger.ex:245]: one more "modulee" -> "module"

File: lib/elixir/lib/config.ex (hunk 1 above)
Comment (josevalim) [reply lib/elixir/lib/config.ex:201]:
```suggestion
  `config/2` or `config/3` call. If `root_key` points to an
```

**Comment by josevalim** [comment]
A potential question we could have here is if we should mirror the functionality in `Application`. If we were to follow their API, we would have:

  * `read_all_config(root_key)`
  * `read_config(root_key, path)`

The reason I chose only a single `read_config/1` is to give more control to the user. Although the path one can be a bit more convenient, I don't foresee many use cases anyway. I want to make it explicitly distinct from the application one (as to avoid any confusion that they are related) and also because they have different performance implications. `get_all_env`, for example, performs multiple ETS reads.

If we want, in the future, we could add: `read_config(root_key, path)` and bang versions for both.

**Review by sabiwara — APPROVED** [review]
LGTM

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**